### PR TITLE
Replace MIP with SciPy

### PR DESCRIPTION
**Pull request by k8culver**
Added on to @arcondello's work

- Replaced MIP with SciPy's MILP HiGHS solver
- Added basic tests for the SciPy HiGHS solver

Super open to feedback on naming choices

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -383,13 +383,13 @@ def run_optimization_cqm(run_click: int, model: str, solver: str, scenario: str,
     cancel=[Input("cancel-button", "n_clicks")],
     prevent_initial_call=True
 )
-def run_optimization_mip(run_click: int,
+def run_optimization_highs(run_click: int,
                          model: str,
                          solver: str,
                          scenario: str,
                          time_limit: int) \
                          -> tuple[go.Figure, go.Figure, str, str]:
```
Comment (arcondello) [inline app.py]:
```suggestion
                           model: str,
                           solver: str,
                           scenario: str,
                           time_limit: int,
                           ) -> tuple[go.Figure, go.Figure, str, str]:
```
incredibly minor, but we should stick with a clean indent.

**Review by arcondello — APPROVED** [review]
LGTM!

File: src/utils/scipy_solver.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        solution = scipy.optimize.milp(c,
+                                       integrality=integrality,
+                                       bounds=scipy.optimize.Bounds(lb=lb, ub=ub),
+                                       options=dict(time_limit=time_limit),
+                                       constraints=constraints,
+                                       )
+        run_time = time.perf_counter() - t
+
+        # If we're infeasible, return an empty solution
+        if not solution.success:
```
Comment (arcondello) [inline src/utils/scipy_solver.py]: I did a bit more testing, and it looks like `not solution.success` when the solver times out, even if it has a valid solution. We should therefore do
```suggestion
        if solution.x is None:
```
which will handle the case that the solver timed out but it has an interim solution.

**Comment by k8culver** [comment]
Closing for now as this repo will be combined with the FSS repo and MIP has already been replaced with SciPy in that repo in this PR https://github.com/dwave-examples/flow-shop-scheduling/pull/1

**Pull request closed without merging**